src/cljs/main/broadfcui/page/workspace/analysis/igv.cljs

```diff
@@ -10,61 +11,58 @@
-                              :headers {"Authorization" (str "Bearer " token)}
-                              :displayMode "EXPANDED"
-                              :height (when bam? 200)
-                              :autoHeight (when bam? false)}))
+                              ;; NB: IGV knows how to infer the track type from its url - but this doesn't work for sourceType="gcs"
+                              ;; so we will set the track type explicitly.
+                              :sourceType "gcs"
+                              :type (case file-extension
+                                      "bam" "alignment"
+                                      "bed" "annotation"
+                                      "vcf" "variant")
+                              :url track-url
+                              :indexURL (when (string? @index-url) @index-url)
+                              ;; NB: we'd love to skip setting indexURL and set indexed=true instead, which allows IGV to search
+                              ;; for indexes by naming convention, similar (but more lazily-loaded than) our track selector modal.
+                              ;; However, IGV's naming convention only supports (filename + ".bai"), e.g. my.bam.bai, whereas
+                              ;; sometimes we also want (filename.replace(".bam", ".bai")), e.g. my.bai.
+                              ;; So, we have keep our own track selector with index-finder.
+                              }))
                          tracks)}))
 
 (react/defc IGVContainer
-  {:component-will-mount
-   (fn []
-     (when-not @igv-styles-loaded?
-       (load-igv-styles)))
+  {
    :render
    (fn [{:keys [this state]}]
      (let [{:keys [deps-loaded? error?]} @state]
        [:div {}
-        [:div {:ref "container"}]
+        [:div {:ref "container" :data-test-id "igv-container"}]
         (cond
           error? (style/create-server-error-message "Unable to load IGV.")
-          deps-loaded? [ScriptLoader
-                        {:key "igv"
-                         :on-error #(swap! state assoc :error? true)
-                         :on-load #(do
-                                     (swap! state assoc :igv-loaded? true)
-                                     (this :refresh))
-                         :path "https://igv.org/web/release/1.0.6/igv-1.0.6.min.js"}]
           :else [ScriptLoader
-                 {:key "igv-deps"
-                  :on-error #(swap! state assoc :error? true)
-                  :on-load #(swap! state assoc :deps-loaded? true)
-                  :path "igv-deps.bundle.js"}])]))
+                  {:key "igv"
+                   :on-error #(swap! state assoc :error? true)
+                   :on-load #(do
+                               (swap! state assoc :igv-loaded? true)
+                               (this :refresh))
+                   :path "https://igv.org/web/release/2.0.0-rc5/dist/igv.min.js"}])]))
```

Comment:
IGV 2.x does support installation via npm. We could do that instead of loading via `ScriptLoader`. I didn't attempt this because:
1) I wanted to keep this PR as small as possible (it's already quite large)
2) I still like only loading the igv js once a user navigates to the relevant tab.

Reply:
Why not install it via npm and load it through the igvDeps bundle? Also, I think that was the only place timingDiagram.js was loaded...?

Reply:
> base-deps.js

No argument here, I just wanted to make sure it was somewhere, I'm not sure why it was in igv to start with.

> follow-on PR

Sure!